Realm/Realm/Realm.cs

```diff
@@ -695,13 +695,23 @@ public Task WriteAsync(Action<Realm> action)
             if (SynchronizationContext.Current != null)
             {
                 var scheduler = TaskScheduler.FromCurrentSynchronizationContext();
+                ulong otherVersion = 0;
                 return Task.Run(() =>
                 {
                     using (var realm = GetInstance(Config))
                     {
                         realm.Write(() => action(realm));
+                        otherVersion = realm.SharedRealmHandle.GetTransactionVersion();
                     }
-                }).ContinueWith(_ => Refresh(), scheduler);
+                }).ContinueWith(_ =>
+                {
+                    if (otherVersion == SharedRealmHandle.GetTransactionVersion())
```

Comment:
Should this be strict equality? I imagine if the current version is higher (for some reason) than the other version, we should still return?